src/com/latacora/backsaws/credential_process.clj

```diff
@@ -0,0 +1,133 @@
+;; This code was copied from the project pod-babashka-aws and then modified by Latacora.
+;;
+;; Original file:
+;; https://github.com/babashka/pod-babashka-aws/blob/81e200692c8b637529cc48f7c51f2e5777c586c5/src/pod/babashka/aws/impl/aws/credentials.clj
+;;
+;; pod-babashka-aws is copyright © 2020 Michiel Borkent, Jeroen van Dijk, Rahul De and Valtteri
+;; Harmainen and is distributed under the Apache License 2.0:
+;; https://github.com/babashka/pod-babashka-aws/blob/81e200692c8b637529cc48f7c51f2e5777c586c5/LICENSE
+;;
+;; Modifications copyright © 2022 Latacora and distributed under the Eclipse Public License 1.0; see
+;; LICENSE in the root of this code repository.
+
+(ns com.latacora.backsaws.credential-process
+  (:require [clojure.data.json :as json]
+            [clojure.java.io :as io]
+            [clojure.java.shell :as shell]
+            [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [cognitect.aws.config :as config]
+            [cognitect.aws.credentials :as creds]
+            [cognitect.aws.util :as u]))
+
+
+(set! *warn-on-reflection* true)
+
+
+(defn- parse-cmd
+  "Split string to list of individual space separated arguments.
+  If argument contains space you can wrap it with `'` or `\"`."
+  [s]
+  (loop [s (java.io.StringReader. s)
+         in-double-quotes? false
+         in-single-quotes? false
+         buf (java.io.StringWriter.)
+         parsed []]
+    (let [c (.read s)]
+      (cond
+        (= -1 c) (if (pos? (count (str buf)))
+                   (conj parsed (str buf))
+                   parsed)
+        (= 39 c) ;; single-quotes
+        (if in-single-quotes?
+        ;; exit single-quoted string
+          (recur s in-double-quotes? false (java.io.StringWriter.) (conj parsed (str buf)))
+        ;; enter single-quoted string
+          (recur s in-double-quotes? true buf parsed))
+
+        (= 92 c) ;; assume escaped quote
+        (let [escaped (.read s)
+              buf (doto buf (.write escaped))]
+          (recur s in-double-quotes? in-single-quotes? buf parsed))
+
+        (and (not in-single-quotes?) (= 34 c)) ;; double quote
+        (if in-double-quotes?
+        ;; exit double-quoted string
+          (recur s false in-single-quotes? (java.io.StringWriter.) (conj parsed (str buf)))
+        ;; enter double-quoted string
+          (recur s true in-single-quotes? buf parsed))
+
+        (and (not in-double-quotes?)
+             (not in-single-quotes?)
+             (Character/isWhitespace c))
+        (recur s in-double-quotes? in-single-quotes? (java.io.StringWriter.)
+               (let [bs (str buf)]
+                 (cond-> parsed
+                   (not (str/blank? bs)) (conj bs))))
+        :else (do
+                (.write buf c)
+                (recur s in-double-quotes? in-single-quotes? buf parsed))))))
```

Comment:
Dear lord am I glad we didn't end up needing to write this.

Reply:
Same!